### ISSUE TITLE
IDA 9.0 support

### ISSUE THREAD

**Issue by lockpicksecurity** [a1ext/labeless#63]
Would there be a release for IDA Pro version 9?

**Comment by a1ext** [comment]
hi @lockpicksecurity, there is work in progress of porting labeless to Python3 and there is one sad bug appeared in the IDA SDK which stops me from releasing new version. I hope the solution will appear soon :) 


--
stay tuned

**Comment by a1ext** [comment]
hi @lockpicksecurity, would you like to take a part in the beta-testing of a new version?

**Comment by lockpicksecurity** [comment]
Hi, yes. Always.


Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by a1ext** [comment]
Here is a pre-release bundle https://drive.google.com/file/d/1hmRjSqvk4kukKRuDRcCLSaMGJbnpJJBs/view?usp=sharing uploaded
Thanks for helping me testing it and feel free to comment here if you spot some issue

**Comment by lockpicksecurity** [comment]
Thanks,

I’ll test out these days and come-up with comments if I spot anything.
Btw, IDA doesn’t ship with 32 bit version anymore (REF: https://docs.hex-rays.com/release-notes/9_0#no-more-ida32).
I mainly use IDA on MacOS with Apple silicon, but historically and occasionally use Win and Linux versions too. Now with the new licensing option I can test anywhere.

**Comment by a1ext** [comment]
> IDA doesn’t ship with 32 bit version anymore (REF: https://docs.hex-rays.com/release-notes/9_0#no-more-ida32).

ohh, I haven't spotted that... That's bad, it breaks some logic how bitness is determined... I will check and update the bundle when I fix that.
Thanks for the hint

**Comment by a1ext** [comment]
Thanks for telling, barely there was just formatting of pointers fixes, I've reuploaded the bundle, link stays the same :)

**Comment by a1ext** [comment]
> I mainly use IDA on MacOS with Apple silicon

I don't have such license, can I ask you some questions, maybe it will be not painfull to add MacOS support too?
Could you check QtCore library version used in the IDA 9th and 8th if you have one?

**Comment by lockpicksecurity** [comment]
Not sure if you mean the QT libraries or something else, but IDA is saying that QT-5 is used. Please let me know if that's what you're looking for. And happy to check elsewhere if you give me a hint what to look for.
________________________________

**Comment by a1ext** [comment]
yes, it is one of shipped Qt libraries dependencies, because I need to prepare a prebuilt Qt 5 of same version.
In the past when I was developing labeless, I appeared IDA haven't had enough UI api to make a better UI as I wanted, so I utilized Qt directly for that XD. for example settings dialog or remote python execution widget with custom editors and completion

**Comment by lockpicksecurity** [comment]
It seems like the QtCore is version 5.15.3 on the latest SP1 IDA 9.0 if that helps.

Also, for the licensing options, not sure if you haven’t seen that, but as long as you have an IDA 9 license the platform no longer matters and the license key can be used in either Win/Lin/Mac
https://docs.hex-rays.com/release-notes/9_0#licensing-changes

**Comment by a1ext** [comment]
Thanks :)
I don't have a license anymore, expired. I'm testing on IDA Free 9 adding python libs/plugin borrowed from my friend...
generally speaking it allows me to test it. if you have IDA 9 for MacOS I would request if you can share archived `/python` folder and an `idapython3*` plugin (for MacOS and for Linux if you have), so that allow me to test corresponding builds using ida demo

**Comment by a1ext** [comment]
I've released new version v [1.1.7.0](https://github.com/a1ext/labeless/releases/tag/v_1_1_7_0)
